app/src/main/java/com/keylesspalace/tusky/components/preference/PreferencesActivity.kt

```diff
@@ -81,27 +84,41 @@ class PreferencesActivity :
                 GENERAL_PREFERENCES -> PreferencesFragment.newInstance()
                 ACCOUNT_PREFERENCES -> AccountPreferencesFragment.newInstance()
                 NOTIFICATION_PREFERENCES -> NotificationPreferencesFragment.newInstance()
-                TAB_FILTER_PREFERENCES -> TabFilterPreferencesFragment.newInstance()
-                PROXY_PREFERENCES -> ProxyPreferencesFragment.newInstance()
                 else -> throw IllegalArgumentException("preferenceType not known")
             }
 
         supportFragmentManager.commit {
             replace(R.id.fragment_container, fragment, fragmentTag)
         }
 
-        when (preferenceType) {
-            GENERAL_PREFERENCES -> setTitle(R.string.action_view_preferences)
-            ACCOUNT_PREFERENCES -> setTitle(R.string.action_view_account_preferences)
-            NOTIFICATION_PREFERENCES -> setTitle(R.string.pref_title_edit_notification_settings)
-            TAB_FILTER_PREFERENCES -> setTitle(R.string.pref_title_post_tabs)
-            PROXY_PREFERENCES -> setTitle(R.string.pref_title_http_proxy_settings)
-        }
-
         onBackPressedDispatcher.addCallback(this, restartActivitiesOnBackPressedCallback)
         restartActivitiesOnBackPressedCallback.isEnabled = savedInstanceState?.getBoolean(EXTRA_RESTART_ON_BACK, false) ?: false
     }
 
+    override fun onPreferenceStartFragment(
+        caller: PreferenceFragmentCompat,
+        pref: Preference
+    ): Boolean {
+        val args = pref.extras
+        val fragment = supportFragmentManager.fragmentFactory.instantiate(
+            classLoader,
+            pref.fragment!!
+        )
+        fragment.arguments = args
+        fragment.setTargetFragment(caller, 0)
+        supportFragmentManager.beginTransaction()
```

Comment:
You can make this nicer with the `commit` extension function. https://developer.android.com/kotlin/ktx#fragment

Reply:
Done.